src/linear_algebra/finite_dimensional.lean

```diff
@@ -149,42 +228,69 @@ end
 @[simp] lemma findim_fin_fun {n : ℕ} : findim K (fin n → K) = n :=
 by simp
 
-variable {K}
-
-section
+/-- The submodule generated by a finite set is finite-dimensional. -/
+theorem span_of_finite {A : set V} (hA : set.finite A) : finite_dimensional K (submodule.span K A) :=
+is_noetherian_span_of_finite K hA
 
-variables {ι : Type w} [fintype ι] [decidable_eq V]
-variables {b : ι → V} (h : is_basis K b)
-include h
-
-lemma fg_of_finite_basis : submodule.fg (⊤ : submodule K V) :=
-⟨ finset.univ.image b, by {convert h.2, simp} ⟩
+end finite_dimensional
 
-lemma finite_dimensional_of_finite_basis : finite_dimensional K V :=
-finite_dimensional.of_fg $ fg_of_finite_basis h
+namespace submodule
+open finite_dimensional
 
-lemma dim_eq_card : dim K V = fintype.card ι :=
-by rw [←h.mk_range_eq_dim, cardinal.fintype_card,
-       set.card_range_of_injective (h.injective zero_ne_one)]
+/-- A submodule is finitely generated if and only if it is finite-dimensional -/
+theorem fg_iff_finite_dimensional (s : submodule K V) :
+  s.fg ↔ finite_dimensional K s :=
+⟨λh, is_noetherian_of_fg_of_noetherian s h,
+ λh, by { rw ← map_subtype_top s, exact fg_map (iff_fg.1 h) }⟩
 
-/-- The findim of a vector space is equal to the cardinality of any basis. -/
-lemma findim_eq_card : by { haveI : finite_dimensional K V := finite_dimensional_of_finite_basis h,
-  exact findim K V = fintype.card ι } :=
+/-- In a finite-dimensional vector space, the dimensions of a submodule and of the corresponding
+quotient add up to the dimension of the space. -/
+theorem findim_quotient [finite_dimensional K V] (s : submodule K V) :
```

Comment:
I find that this name doesn't really cover the statement... `findim_quotient_add_findim`? Maybe a bit long...

Reply:
I copied the name from `dimension.lean`, where it is called `dim_quotient`. I agree that the name is bad, and I will change both.